### ISSUE TITLE
ggplot2 crashing recent r-devel builds

### ISSUE THREAD

**Issue by briencj** [tidyverse/ggplot2#2336]
With "R Under development (unstable) (2017-06-27 r72859)", the reprex shows that ggplot produces a plot that is a grid of numbers. However, under the latest "R Under development (unstable) (2017-11-11 r73707)" build, the reprex crashes the R console, as shown in the attached image. As I result, I have been unable to compile packages (dae - https://cran.r-project.org/package=dae and imageData - https://cran.r-project.org/package=imageData) that depend on ggplots with the latest version of r-devel, even though they compile successfully with build r72859. 
![ggplottest73707](https://user-images.githubusercontent.com/12837653/32697189-508c37be-c7db-11e7-93f6-f49bcacfcd0e.png)
 
Cheers, Chris Brien

---

This code crashes the latest development versions of R

```r
``` r
library(ggplot2)
(RCBD.lay <- data.frame(Blocks = factor(rep(1:5, each = 6)), 
                        Units = factor(rep(1:6, times = 5)),
                        Treatments = factor(rep(1:6, times = 5))))
#>    Blocks Units Treatments
#> 1       1     1          1
#> 2       1     2          2
#> 3       1     3          3
#> 4       1     4          4
#> 5       1     5          5
#> 6       1     6          6
#> 7       2     1          1
#> 8       2     2          2
#> 9       2     3          3
#> 10      2     4          4
#> 11      2     5          5
#> 12      2     6          6
#> 13      3     1          1
#> 14      3     2          2
#> 15      3     3          3
#> 16      3     4          4
#> 17      3     5          5
#> 18      3     6          6
#> 19      4     1          1
#> 20      4     2          2
#> 21      4     3          3
#> 22      4     4          4
#> 23      4     5          5
#> 24      4     6          6
#> 25      5     1          1
#> 26      5     2          2
#> 27      5     3          3
#> 28      5     4          4
#> 29      5     5          5
#> 30      5     6          6
(ggplot(data = RCBD.lay, aes(x = Units, y = Blocks, label = Treatments)) +
    geom_text(aes(colour = Treatments)))
```

![](https://i.imgur.com/gMUYvkx.png)

**Comment by hadley** [comment]
Unfortunately this is unlikely to be a problem with ggplot2 but instead represents a mismatch in installed package versions. Try reinstalling ggplot2 and all its dependencies, and if that doesn’t help you might need to delete and reinstall your entire library.

**Comment by briencj** [comment]
I did as you suggested. Other than the packages that come with base R, all of the packages that I have installed are in a library tree on a separate hard disk partition. I deleted this library and reinstalled the entire library from CRAN using the development build of 2017-11-10 r73707 running in RStudio 1.1.383. I have ggplot2 version 2.2.1 installed. 

For me, the reprex still crashes the R console, or, if run the  from Rstudio, the session is aborted.

Just to be certain, are you saying that you have successfully run the reprex on an installation with a recent (last 2 weeks) build of r-devel?  (As I mentioned above, I know that it works on a build from June.) 

Cheers, Chris
